h/static/scripts/directive/test/annotation-test.js

```diff
@@ -0,0 +1,1377 @@
+          id: 'deadbeef',
+          document: {
+            title: 'A special document'
+          },
+          target: [{}],
+          uri: 'http://example.com',
+          user: 'acct:bill@localhost'
+        };
+      }
+    )
+  );
+
+  afterEach(function() {
+    sandbox.restore();
+  });
+
+  describe('when the annotation is a highlight', function() {
+    beforeEach(function() {
+      annotation.$highlight = true;
+      annotation.$create = sinon.stub().returns({
+        then: angular.noop,
+        'catch': angular.noop,
+        'finally': angular.noop
+      });
+    });
+
+    it('persists upon login', function() {
+      delete annotation.id;
+      delete annotation.user;
+      fakeSession.state.userid = null;
+      createDirective();
+      $scope.$digest();
+      assert.notCalled(annotation.$create);
+      fakeSession.state.userid = 'acct:ted@wyldstallyns.com';
+      $scope.$broadcast(events.USER_CHANGED, {});
+      $scope.$digest();
+      assert.calledOnce(annotation.$create);
+    });
+
+    it('is private', function() {
+      delete annotation.id;
+      createDirective();
+      $scope.$digest();
+      assert.deepEqual(annotation.permissions, {
+        read: ['justme']
+      });
+    });
+  });
+
+  describe('#reply', function() {
+    var container;
+
+    beforeEach(function() {
+      createDirective();
+      annotation.permissions = {
+        read: ['acct:joe@localhost'],
+        update: ['acct:joe@localhost'],
+        destroy: ['acct:joe@localhost'],
+        admin: ['acct:joe@localhost']
+      };
+    });
+
+    it('creates a new reply with the proper uri and references', function() {
+      var match;
+      controller.reply();
+      match = sinon.match({
+        references: [annotation.id],
+        uri: annotation.uri
+      });
+      assert.calledWith(fakeAnnotationMapper.createAnnotation, match);
+    });
+
+    it('makes the annotation shared if the parent is shared', function() {
+      var reply;
+      reply = {};
+      fakeAnnotationMapper.createAnnotation.returns(reply);
+      fakePermissions.isShared.returns(true);
+      controller.reply();
+      assert.deepEqual(reply.permissions, {
+        read: ['everybody']
+      });
+    });
+
+    it('makes the annotation shared if the parent is shared', function() {
+      var reply;
+      $scope.annotation.group = 'my group';
+      $scope.annotation.permissions = {
+        read: ['my group']
+      };
+      reply = {};
+      fakeAnnotationMapper.createAnnotation.returns(reply);
+      fakePermissions.isShared = function(permissions, group) {
+        return permissions.read.indexOf(group) !== -1;
+      };
+      fakePermissions.shared = function(group) {
+        return {
+          read: [group]
+        };
+      };
+      controller.reply();
+      assert(reply.permissions.read.indexOf('my group') !== -1);
+    });
+
+    it(
+      'does not add the world readable principal if the parent is private',
+      function() {
```

Comment:
Small formatting nit. For all hand-written JS we have a space between the `function` keyword and the parenthesies in function expressions. We should be be consistent in these files as well.